src/storage/storage_manager.cpp

```diff
@@ -61,27 +61,45 @@ bool ObjectCache::ObjectCacheEnabled(ClientContext &context) {
 }
 
 idx_t StorageManager::GetWALSize() {
-	auto wal_ptr = GetWAL();
+	auto wal_ptr = GetWALIfExists();
 	if (!wal_ptr) {
 		return 0;
 	}
 	return wal_ptr->GetWALSize();
 }
 
+optional_ptr<WriteAheadLog> StorageManager::GetWALIfExists() {
+	if (InMemory() || read_only || !load_complete) {
+		return nullptr;
+	}
+
+	if (!wal && wal_file_might_exists) {
+		auto wal_path = GetWALPath();
+		auto &fs = FileSystem::Get(db);
+		if (fs.FileExists(wal_path)) {
+			wal = make_uniq<WriteAheadLog>(db, wal_path);
```

Comment:
There's also logic for checking if the WAL exists in the `WriteAheadLog` class itself - I think it would be better to keep that logic contained there instead of having this logic both in the storage manager and in the WAL class